### PR TITLE
chore(deps): updated to the latest versions

### PR DESCRIPTION
**Pull request by easen-amp**
* upgraded everything using `npm-check-updates` 
* fixed prettier issues
* updated snapshots (jest upgrade has added some new props)
* added lint-staged instead of using a pretest hook to lint out changes

### CODE REVIEW

File: common/services/blog-post.service.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { defaultClientConfig } from './dynamic-content-client-config';
 export async function parseContent(content: AmplienceContent[]): Promise<AmplienceContent[]> {
   const updatedContent: AmplienceContent[] = [];
 
-  for (let c of content) {
+  for (const c of content) {
```
Comment (easen-amp) [inline common/services/blog-post.service.ts:12]: manual change - eslint flagged this up

File: common/services/blog-reference-list.service.spec.ts [hunk 2]
```diff
@@ -26,8 +26,8 @@ describe('getBlogReferenceList', (): void => {
         name: 'test',
         deliveryId: 'test-delivery-id',
         schema: 'test-schema',
-        toJSON: (): void => {}
-      }
+        toJSON: jest.fn(),
```
Comment (easen-amp) [inline common/services/blog-reference-list.service.spec.ts:29]: eslint doesn't like empty functions, so switched to use jest mocks

File: common/services/blog-date.service.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ const MONTHS = [
   'September',
   'October',
   'November',
-  'December'
+  'December',
```
Comment (simonmudd) [inline common/services/blog-date.service.ts:13]: trailing comma?

**Comment by easen-amp** [comment]
> Trailing commas everywhere, not sure this is intentional?

Prettier 2.0, trailing commas is now turned on by default

**Pull request closed without merging**